### ISSUE TITLE
QUERY: `ndimage.binary_fill_holes` doesn't fill all holes

### ISSUE THREAD

**Issue by windyjunfeng** [scipy/scipy#21500]
Hi, I have a question about the result of `scipy.ndimage.binary_fill_holes`. The following binary 3D image is the output result, but there are some closed holes. I want to set all parts surrounded by the foreground to 1 like the second figure. Could you please give some suggestions? Thank you very much!
![a957249ec5dc1170bba95cfec302bb0](https://github.com/user-attachments/assets/6e3770d1-0141-4bce-bfd6-50611d87843b)
![image](https://github.com/user-attachments/assets/d7baa5fc-669f-47fb-8206-64fd5b7a9e3f)

**Comment by lucascolley** [comment]
Sorry @windyjunfeng , I'm not familiar with this function, so I can't answer your question. It looks like it was written and documented around 20 years ago and hasn't been changed much since!

**Comment by lucascolley** [comment]
One suggestion I have if you don't receive a reply here is to ask your question on [the User Discussion section of the Scientific Python Discourse](https://discuss.scientific-python.org/c/user/17). That might get a few more eyes from some scikit-image people, and perhaps there is an alternative solution available there, or they can otherwise help.

**Comment by windyjunfeng** [comment]
Thank you! I saved the website. I solved it just now. I made it by setting `origin` to (0,1,0), (0,1,0) (inverse the image along y axis at the same time) and (0,0,1) respectively and computing the intersection. I think the parameter `origin` is the initial position of the structure element, so different positions may lead to different results. It depends on the image type. If there is no complex holes and foreground in the image, just using the default parameter once will be OK.
